packages/dai-plugin-governance/src/SpellService.js

```diff
@@ -0,0 +1,96 @@
+import { PublicService } from '@makerdao/services-core';
+import { PAUSE } from './utils/constants';
+import DsSpellAbi from '../contracts/abis/DSSpell.json';
+import padStart from 'lodash/padStart';
+import assert from 'assert';
+import contractInfo from '../contracts/contract-info.json';
+const pauseInfo = contractInfo.pause;
+import { netIdToName } from './utils/helpers';
+
+export default class SpellService extends PublicService {
+  constructor(name = 'spell') {
+    super(name, ['smartContract', 'web3']);
+    this.eta = {};
+    this.done = {};
+    this.action = {};
+    this.executionDate = {};
+  }
+
+  getDelayInSeconds() {
+    if (this.delay) return this.delay;
+    this.delay = this._pauseContract().delay();
+    return this.delay;
+  }
+
+  async getEta(spellAddress) {
+    if (this.eta[spellAddress]) return this.eta[spellAddress];
+    const spell = this.get('smartContract').getContractByAddressAndAbi(
+      spellAddress,
+      DsSpellAbi
+    );
+    const eta = await spell.eta();
+    assert(eta, `eta has not yet been set for spell ${spellAddress}`);
```

Comment:
do you know if a call to the wrapped smart contract will return falsy if the value isn't set? I only ask b/c i know `eth_call`s will just return a `0`

Reply:
Woops, I assumed it returned 0, which is falsy, but it returns BigNumber(0), which isn't falsy. I think I'll just change this to return false, rather than throw an error, if it's not set.